### ISSUE TITLE
Script fails if git information global instead local

### ISSUE THREAD

**Issue by Jey-Cee** [AlCalzone/release-script#81]
If the script runs in other user context as the logged in user it is necessary to set the user.email & user.name local.
The dropped message says set it global, but in this case this will not help.

As a workaround the script could ask for this information and go ahead.

OS: Ubuntu

**Comment by AlCalzone** [comment]
I'm not sure if/how it is possible to detect your situation programmatically. I'm still confused why `git` cannot access the global info if you are able to work in the other user's directory.

What is the output if you execute these commands (please obfuscate sensitive data if there is any)?
```
git config --list --global
git config --list --local
git config --list --show-origin
```

**Comment by Jey-Cee** [comment]
Here is the output how it looks now while it is working 

git config --list --global:
`user.name=Jey Cee`
`user.email=jey-cee@live.com`
`core.autocrlf=input`
`credential.helper=store`


git config --list --local:
`core.repositoryformatversion=0`
`core.filemode=true`
`core.bare=false`
`core.logallrefupdates=true`
`remote.origin.url=https://github.com/Jey-Cee/ioBroker.enocean.git`
`remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*`
`branch.master.remote=origin`
`branch.master.merge=refs/heads/master`
`branch.smart_ACK.remote=origin`
`branch.smart_ACK.merge=refs/heads/smart_ACK`
`branch.ser2net.remote=origin`
`branch.ser2net.merge=refs/heads/ser2net`
`user.email=jey-cee@live.com`
`user.name=Jey Cee`


git config --list --show-origin:
`file:/home/christoph/.gitconfig user.name=Jey Cee`
`file:/home/christoph/.gitconfig user.email=jey-cee@live.com`
`file:/home/christoph/.gitconfig core.autocrlf=input`
`file:/home/christoph/.gitconfig credential.helper=store`
`file:.git/config        core.repositoryformatversion=0`
`file:.git/config        core.filemode=true`
`file:.git/config        core.bare=false`
`file:.git/config        core.logallrefupdates=true`
`file:.git/config        remote.origin.url=https://github.com/Jey-Cee/ioBroker.enocean.git`
`file:.git/config        remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*`
`file:.git/config        branch.master.remote=origin
`file:.git/config        branch.master.merge=refs/heads/master`
`file:.git/config        branch.smart_ACK.remote=origin`
`file:.git/config        branch.smart_ACK.merge=refs/heads/smart_ACK`
`file:.git/config        branch.ser2net.remote=origin`
`file:.git/config        branch.ser2net.merge=refs/heads/ser2net`
`file:.git/config        user.email=jey-cee@live.com`
`file:.git/config        user.name=Jey Cee`

**Comment by Jey-Cee** [comment]
Here is the output of a new adapter

git config --list --local:
`core.repositoryformatversion=0`
`core.filemode=true`
`core.bare=false`
`core.logallrefupdates=true`
`remote.origin.url=https://github.com/Jey-Cee/ioBroker.testing.git`
`remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*`
`branch.master.remote=origin`
`branch.master.merge=refs/heads/master`

**Comment by AlCalzone** [comment]
Seems that `npm` changes the user's context of the executed script to the one that owns the folder, so the global config would be in a different folder than when executed by hand.
I should be able to detect this by reading the `user.name` and `user.email` and issue a warning.

**Comment by AlCalzone** [comment]
fixed in v2.2.1 - should be out any minute
